lua/lspconfig/util.lua

```diff
@@ -68,6 +68,27 @@ function M.insert_package_json(config_files, field, fname)
   return config_files
 end
 
+--- Appends mix.exs files to the `config_files` list if `field`
+--- is found in any such file in any ancestor of `fname`.
+---
+--- NOTE: this does a "breadth-first" search, so is broken for multi-project workspaces:
+--- https://github.com/neovim/nvim-lspconfig/issues/3818#issuecomment-2848836794
+function M.insert_mix_exs(config_files, field, fname)
```

Comment:
We're trying to eliminate util.lua, not grow it.

As a compromise, I will merge this and followup with a PR that generalizes `insert_package_json`, it has no reason to hardcode `'package.json', 'package.json5'` ...